### ISSUE TITLE
Add an additional search parameter to `simulate`

### ISSUE THREAD

**Issue by konnov** [informalsystems/apalache#2143]
As raised in #2139, it would be good to extend `simulate` with an additional option `--max-error-per-run=k` that changes the behavior of the search in conjunction with `--max-error=m` as follows:

 - For a randomly generated prefix of actions, the model checker keeps finding errors until `k` errors are found for the prefix, or `m` errors were generated in total (for all prefixes visited so far).
 - Once `k` errors were found for a prefix, switch to another prefix.
 - If `m` errors were found in total, terminate the search.

@andrey-kuprianov, is it what you had in mind in #2139?

@andrey-kuprianov, @tesnimab, what is the priority of this issue? We planned to work on the symbolic simulator in Q4.

**Comment by thpani** [comment]
Given the outgrowth of CLI flags, maybe there is a default we can implement first?
We can always expose more configurable behavior if it's requested by the users.

**Comment by andrey-kuprianov** [comment]
@konnov yes, I think this is very reasonable. It is not really urgent, rather a nice-to-have from my point of view. My guess is that it's probably also better to wait for the full-fledged server mode before diving in here, as the server mode should provide better means for exploration? Only guessing here.

**Comment by konnov** [comment]
With the server mode, you would need us to implement anything in the addition. You would just implement your exploration strategy on top of the client library.
